### PR TITLE
#4441: Improve animations and dynamically import animation libraries to reduce bundle size

### LINKED ISSUE #4441: Eliminate dependency on Velocity
## Issue Description
We currently have a dependency on `Velocity` for animating parts of our client application.
Velocity is a DOM animation library which makes doing some complex animations performant and simple.
We currently use velocity to do one of two things.
1. Animate the cart drawer opacity
2. Animate the background color of changed fields from green and then back to white.

Neither of these animations should need a library to do well or simply and don't justify the added overhead of [94.5kb](https://bundlephobia.com/result?p=velocity-react@1.4.1)

The animations using velocity are contained in one of 6 files
- containers/cartIconContainer.js
- components/metadata/metafield.js
- components/tagItem.js
- components/productAdmin.js
- components/productField.js
- components/variantForm.js

This issue is to eliminate the dependency on Velocity and replace any animations that are using Velocity with another method of animation.

### PR DESCRIPTION
**Pull request by dancastellon**
Resolves #4441 
Impact: **minor**  
Type: **performance**

## Issue
We currently have `velocity-animate` in the client bundle at 94.5kb. We use Velocity for 5 things:
1) Animate the cart drawer
2) Flash a green color when input fields change
3) Animate the sliding out of the admin panel
4) Animate sliding up & down of admin panel cards
5) Animating the opacity of the overlay when the orders panel is opened

## Solution
Most of the animations are replaceable with CSS. Remove `velocity-animate` and replace those, using a new component called `CSSTransitionGroup` and CSS. Animating the height of admin panel cards requires JS. Use the small `react-animate-height` library for that.

Whether using CSSTransitionGroup or Animate height, the bundle size should not be affected (dynamic imports).

Note: This PR removes `velocity-animate` and `velocity-react` from package.json, and adds `react-transition-group` and `react-animate-height` - both of which are much smaller then Velocity.

## Breaking changes
None

## Testing
1. Visit the homepage and open the cart drawer - confirm improved animation
2. Click a tag in the nav and confirm cart drawer is still open
3. Confirm on tag pages you can toggle cart drawer
4. Change an existing tag's title in the header, confirm that leaving the input after change, or pressing enter after change, the input is flashed green
5. PDP -> Details - Edit a metafield's key, confirm it flashes green. Confirm same with value
6. Edit the product's title inline, and confirm it flashes green leaving the input after change, or pressing enter after change
7. Edit subtitle inline and confirm it flashes green
8. Edit vendor field inline and confirm it flashes green
9. Edit a field in the variant form (other than tax code). Confirm it flashes a lighter green color
10. Confirm admin panel can slide open/closed
11. Confirm admin panel cards can slide up/down
12. Click "Orders" and confirm black overlay fades in
13. Click through to order detail, confirm detail panel slides in
14. Confirm 10-13 while using an RTL language
15. Confirm switching from LTR to RTL language, the admin bar is positioned correctly (no empty scrollbar on RTL, no cut-off icons when switching back to LTR)
16. Confirm double-clicking a product in grid - in edit mode - brings you to PDP without crashing the app or any errors in console.
17. Confirm none of these packages are in the client bundle: `velocity-animate`, `velocity-react`, `react-animate-height`, `react-transition-group`. Run with `export TOOL_NODE_FLAGS='--max-old-space-size=4096' && METEOR_DISABLE_OPTIMISTIC_CACHING=1 meteor run --production --extra-packages bundle-visualizer`

## Before
velocity-before

## After
velocity-after

### CODE REVIEW

File: imports/plugins/core/ui/client/components/cards/cardBody.js [hunk 1]
```diff
@@ -47,6 +51,21 @@ class CardBody extends Component {
   }
 
   render() {
+    const { VelocityTransitionGroup } = this.state;
+    if (VelocityTransitionGroup === undefined) {
+      import("velocity-react")
```
Comment (zenweasel) [inline imports/plugins/core/ui/client/components/cards/cardBody.js]: Why aren't eliminating this as well?

**Comment by zenweasel** [comment]
@dancastellon When editing tags I see the green flash when exiting the input but not when entering

**Comment by zenweasel** [comment]
Seeing this error when I double-click on a product logged in as an admin
```
TypeError: Cannot read property 'add' of undefined
    at highlightInput (animations.js:11)
    at ProductAdmin.animateFieldFlash (productAdmin.js:188)
    at ProductAdmin.UNSAFE_componentWillReceiveProps (productAdmin.js:145)
    at callComponentWillReceiveProps (modules.js?hash=1106023c7619d36eb7b0e0c97f1437b533dbe087:200916)
    at updateClassInstance (modules.js?hash=1106023c7619d36eb7b0e0c97f1437b533dbe087:201105)
    at updateClassComponent (modules.js?hash=1106023c7619d36eb7b0e0c97f1437b533dbe087:202539)
    at beginWork (modules.js?hash=1106023c7619d36eb7b0e0c97f1437b533dbe087:203210)
    at performUnitOfWork (modules.js?hash=1106023c7619d36eb7b0e0c97f1437b533dbe087:205249)
    at workLoop (modules.js?hash=1106023c7619d36eb7b0e0c97f1437b533dbe087:205288)
    at renderRoot (modules.js?hash=1106023c7619d36eb7b0e0c97f1437b533dbe087:205328)
```

**Comment by dancastellon** [comment]
@zenweasel I originally left the `<VelocityTransitionGroup />` component in because it had animations that triggered when components mounted/unmounted - something I thought would be tricky to replace with pure JS & CSS. After some research I found `<ReactCSSTransitionGroup />`, which allows me to use CSS animations on mount/unmount. It worked for all but 1 animation, the slide up/down on the dashboard panels. For that, I found `<AnimateHeight />`, and it works smoothly.

The downside is that while this would let us remove `velocity-animate` and `velocity-react` from package.json, we'd be adding 2 more libs - `react-transition-group` and `animate-height`. I'm wondering if you think it's worth it, or if we should revert back to keeping just `velocity-react` for the `<VelocityTransitionGroup />` which we'd dynamically import (so the client bundle size isn't affected).

**Comment by zenweasel** [comment]
@dancastellon What's the status on this? Can we quickly make the changes that @aldeed requested?

**Comment by dancastellon** [comment]
@zenweasel This one's ready to go. This morning I fixed a `Can't call setState on an unmounted component` error and ran through the testing instructions.

**Pull request merged**